### PR TITLE
Tidy test helpers

### PR DESCRIPTION
**Pull request by kanes115**
This PR extracts common functions to helpers so that no SUITE calls another's SUITE's function.

### CODE REVIEW

**Comment by kanes115** [comment]
TODO:
- [ ] export only public functions (in modules that have `-compile(export_all)`

**Comment by kanes115** [comment]
When deleting user, the user cannot be authorised now. It manifests in end_per_groups.

**Comment by kanes115** [comment]
IMPORTANT: Before merging we need to merge changes in escalus and chage rebar.config in mongoose's tests accordingly !!!

File: big_tests/tests/mod_global_distrib_SUITE.erl [hunk 1]
```diff
@@ -586,8 +587,8 @@ test_pm_with_ungraceful_reconnection_to_different_server(Config0) ->
               escalus:assert(is_chat_message, [<<"Hi from Europe1!">>], FromAlice),
               escalus:assert(is_chat_message, [<<"Hi again from Europe1!">>], AgainFromAlice),
               escalus:assert(is_chat_message, [<<"Hi from Asia!">>], AgainFromEve)
-      end),
-    escalus_users:delete_users(Config, [{eve, [{port, 5222} | EveSpec]}]).
```
[the remark this answers is not in the document]
Comment (kanes115) [reply big_tests/tests/mod_global_distrib_SUITE.erl]: Escalus fresh deletes users on its own as well in end_per_*. This line caused some trouble with double-deleting user.

File: big_tests/tests/mongoose_helper.erl [hunk 2]
```diff
@@ -21,8 +21,10 @@
 -export([ensure_muc_clean/0]).
 -export([successful_rpc/3]).
 -export([logout_user/2]).
+-export([wait_until/3, wait_until/4]).
 
 -define(RPC(M, F, A), escalus_ejabberd:rpc(M, F, A)).
```
Comment (fenek) [inline big_tests/tests/mongoose_helper.erl]: This macro is not used now, right?

Comment (kanes115) [reply big_tests/tests/mongoose_helper.erl]: sure

**Comment by kanes115** [comment]
@fenek ping

**Comment by fenek** [comment]
You've mentioned "export only public functions (in modules that have `-compile(export_all)`" as one of TODOs. Some helper modules still use it. Do you plan to improve it as well or would you prefer to leave it for next sprint?

**Comment by kanes115** [comment]
@fenek, I would prefer not to loose focus on my current tasks and eradicate those export_alls in the next sprint. How about that?

**Pull request merged**